services/121-service/src/utils/send-xlsx-response.ts

```diff
@@ -19,14 +25,23 @@ export function sendXlsxReponse(
 }
 
 function arrayToXlsx(array: any[]): Buffer {
-  const worksheet: XLSX.WorkSheet = XLSX.utils.json_to_sheet(array, {
-    dense: true,
+  const workbook = XLSX.utils.book_new();
+
+  let dataSheet = 1;
+  for (let index = 0; index < array.length; index += CHUNK_SIZE) {
+    const chunk = array.slice(index, index + CHUNK_SIZE);
+    const worksheet: XLSX.WorkSheet = XLSX.utils.json_to_sheet(chunk, {
+      dense: true,
+    });
+    XLSX.utils.book_append_sheet(workbook, worksheet, `data_${dataSheet++}`);
+  }
```

Comment:
The performance bottle-neck seem to be here with "parsing the array into a rows-in-a-sheet", right?

And the 'workaround' is to have multiple sheets in the workbook.

Would it also be helping (memory-wise) to do the conversion from JSON-to-sheet in a loop and append all to the same worksheet?

(Following some example-code from: https://docs.sheetjs.com/docs/api/utilities/array#array-of-objects-input )

So, something like:
```suggestion
  const workbook = XLSX.utils.book_new();
  const worksheet = XLSX.utils.aoa_to_sheet([[]]); // Empty worksheet

  array.forEach((row) => {
    XLSX.utils.sheet_add_json(worksheet, row, {
      origin: -1,
    });
  });
  
  XLSX.utils.book_append_sheet(workbook, worksheet);
```

(👆 Not tested in any way.)

Reply:
Let me test it.

Reply:
Its not helping (not working, giving error on `Object.sheet_add_json`), not sure whats the actual reason.

Reply:
Could have something to do with how the sheet-headers are handled?

It the example they 'manually' add the first row etc...